pkg/syncer/cnsoperator/controller/cnsregistervolume/cnsregistervolume_controller.go

```diff
@@ -505,6 +512,116 @@ func (r *ReconcileCnsRegisterVolume) Reconcile(ctx context.Context,
+			storagePolicyUsageCR := &storagepolicyusagev1alpha1.StoragePolicyUsage{}
+			patchedStoragePolicyUsageCR := &storagepolicyusagev1alpha1.StoragePolicyUsage{}
+			if len(storagePolicyUsageList.Items) > 0 {
+				for _, item := range storagePolicyUsageList.Items {
+					if item.Spec.StoragePolicyId == volume.StoragePolicyId && item.Spec.StorageClassName == storageClassName {
+						log.Debugf("Found storagePolicyUsage CR with matching "+
+							"storagePolicyId: %q & storageClassName: %q in namespace: %q", volume.StoragePolicyId, storageClassName,
+							namespace)
+						storagePolicyUsageCR = item.DeepCopy()
+						foundStoragePolicyUsageCR = true
+						break
+					}
+				}
+
+				if foundStoragePolicyUsageCR {
+					// Patch an increase of "reserved" in storagePolicyUsageCR.
+					patchedStoragePolicyUsageCR = storagePolicyUsageCR.DeepCopy()
+					if storagePolicyUsageCR.Status.ResourceTypeLevelQuotaUsage != nil {
+						patchedStoragePolicyUsageCR.Status.ResourceTypeLevelQuotaUsage.Reserved.Add(
+							*resource.NewQuantity(capacity.Value(), capacity.Format))
+					} else {
+						var (
+							usedQty     resource.Quantity
+							reservedQty resource.Quantity
+						)
+						reservedQty = *resource.NewQuantity(capacity.Value(), capacity.Format)
+						patchedStoragePolicyUsageCR.Status = storagepolicyusagev1alpha1.StoragePolicyUsageStatus{
+							ResourceTypeLevelQuotaUsage: &storagepolicyusagev1alpha1.QuotaUsageDetails{
+								Reserved: &reservedQty,
+								Used:     &usedQty,
+							},
+						}
+					}
+					err := patchStoragePolicyUsage(ctx, cnsOperatorClient, storagePolicyUsageCR,
+						patchedStoragePolicyUsageCR)
+					if err != nil {
+						log.Errorf("patching operation failed for StoragePolicyUsage CR: %q in namespace: %q. err: %v",
+							storagePolicyUsageCR.Name, storagePolicyUsageCR.Namespace, err)
+					}
+					// Retrieve the CR
+					currentStoragePolicyUsageCR := &storagepolicyusagev1alpha1.StoragePolicyUsage{}
+					finalStoragePolicyUsageCR := &storagepolicyusagev1alpha1.StoragePolicyUsage{}
+					key := apitypes.NamespacedName{Namespace: storagePolicyUsageCR.Namespace,
+						Name: storagePolicyUsageCR.Name}
+					err = cnsOperatorClient.Get(ctx, key, currentStoragePolicyUsageCR)
+					if err != nil {
+						log.Errorf("failed to get %s CR from supervisor namespace %q. Error: %+v",
+							storagePolicyUsageCR.Name, storagePolicyUsageCR.Namespace, err)
+						return reconcile.Result{RequeueAfter: timeout}, nil
+					}
+					finalStoragePolicyUsageCR = currentStoragePolicyUsageCR.DeepCopy()
+					// Decrease the Reserved field for StoragePolicyUsageCR
+					finalStoragePolicyUsageCR.Status.ResourceTypeLevelQuotaUsage.Reserved.Sub(
+						*resource.NewQuantity(currentStoragePolicyUsageCR.Status.ResourceTypeLevelQuotaUsage.Reserved.Value(),
+							currentStoragePolicyUsageCR.Status.ResourceTypeLevelQuotaUsage.Reserved.Format))
+					// Increase the Used field for StoragePolicyUsageCR
```

Comment:
Didn't we decide on directly updating used field during static volume provisioning? Why increase reserved and then immediately decrease reserved/increase used?

Reply:
See comment here: https://github.com/kubernetes-sigs/vsphere-csi-driver/pull/2697#discussion_r1425560885

Reply:
Resolving the comment based on Xing's review comment